src/Server/AuthorizationMiddleware.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrOAuth2\Server;
+
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Zend\Stratigility\MiddlewarePipe;
+
+/**
+ * Middleware for handling the authorization logic
+ *
+ * This middleware currently create three URLs, based on the OAuth specification:
+ *
+ *      - /oauth/authorize: NOT IMPLEMENTED YET
+ *      - /oauth/token: generate an access token
+ *      - /oauth/revoke: revoke an existing token
+ */
+class AuthorizationMiddleware extends MiddlewarePipe
+{
+    /**
+     * @var AuthorizationServerInterface
+     */
+    private $authorizationServer;
+
+    /**
+     * @param AuthorizationServerInterface $authorizationServer
+     */
+    public function __construct(AuthorizationServerInterface $authorizationServer)
+    {
+        parent::__construct();
+
+        $this->authorizationServer = $authorizationServer;
+
+        $this->pipe('/authorize', [$this, 'handleAuthorizeRequest']);
```

Comment:
@weierophinney: how could I specify a specific method here? Should I actually extends "Zend\Expressive\Application" instead of Stratigility middleware, and use $this->get/post instead ?